### DISCUSSION TITLE
Detect Mobile Context

### DISCUSSION THREAD

**Discussion by scwunch** [SilentVoid13/Templater#652] [Help] [answered]
I'm sure there are many ways of doing this, but what is the best way to detect, in a templater script, whether that script is running on mobile or on desktop?

**Comment by welpdx** [comment, marked as answer]
You can access this info via the [Platform function](https://github.com/obsidianmd/obsidian-api/blob/9bb30633586c6649474b2affdf5b87efdd86498f/obsidian.d.ts#L2326). Since Templater has imported this function from the api, you can access via `tp.obsidian.Platform` (see[ templater docs on the tp.obsidian module](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/obsidian-module.html)) :

```js
<%* console.log(tp.obsidian.Platform["isDesktop"]) %>
<%* console.log(tp.obsidian.Platform["isMobile"]) %>
```
```console
true
false
```

---
This is what Platform returns:
```json
{
    "isDesktop": true,
    "isMobile": false,
    "isDesktopApp": true,
    "isMobileApp": false,
    "isIosApp": false,
    "isAndroidApp": false,
    "isMacOS": false,
    "isSafari": false
}
```

Hope that helps. 

~Welp

Edit: Clarifications

**Reply by scwunch** [reply]
Thank you!  That is more elegant than what I came up with using the `basePath` property of the vault adapter: 
```
<%*
if (app.vault.adapter.basePath.startsWith("/storage/emulated/0"))
    tR += "Mobile";
else
    tR += "Desktop";
%>
```

**Reply by welpdx** [reply]
That is a cool approach as well!
